packages/jest-editor-support/src/SnapshotSupport.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+'use strict';
+
+const babylon = require('babylon');
+const fs = require('fs');
+const path = require('path');
+const walker = require('./walker');
+
+type Node = any;
+
+type SnapshotMetadata = {
+  exists: true | false,
+  exportName: string,
+  node: Node,
+  content?: string,
+};
```

Comment:
Oh and yes exists can be false the first time you're writing the test as there's no snapshot yet.